### DISCUSSION TITLE
Intermittent "Stream not found" and "Missing query" errors

### DISCUSSION THREAD

**Discussion by shankj3** [enisdenjo/graphql-sse#70] [Q&A]
Hey everybody, 

We just started using graphql-sse for subscriptions, and for the most part it is working well. The flow for our subscription, which is to serve updates to a filterable table of data is:
1. On page load, a subscription is created on whatever the current filters are. 
2. If the filters are changed, the existing subscription is cancelled, and a new one is created with the new filter set. 

It is always the same graphql subscription document, just the filter inputs can be changed. 

We are hitting some intermittent errors, where we get `{"errors":[{"message":"Missing query"}]}` (400) or `{"errors":[{"message":"Stream not found"}]}` (404) on the POST / GET calls that are made, which will resolve themselves either on a retry or after refreshing the page (sometimes a few times). 

Has anyone hit this before or know what could be happening? I haven't had much luck looking around on the internet / the issues that have been raised here. 

Using `2.1.3` for client and server. 

Thanks!

**Comment by enisdenjo** [comment]
Hey there, thanks for writing! Sad to hear you're experiencing issues.

To start off, I'd like to confirm that you're using the "single connection mode" ([`singleConnection: true` on the client](https://the-guild.dev/graphql/sse/docs/interfaces/client.ClientOptions#singleconnection))? Do you use it for all the subscriptions?

Seems like we're dealing with 2 different issues here:
- "Missing query" which is a request parsing issue where the `query` parameter of the operation request is missing.
- "Stream not found" which is a "single connection mode" specific issue meaning that an operation request couldn't find the relevant SSE stream (either because it wasnt registered or it was prematurely deleted)

I'd like to gain more insight into your problem, do you mind sharing a repro I can play around with? Looking forward to your reply, thanks!
